src/codegen/compile.ml

```diff
@@ -7206,6 +7206,39 @@ module Serialization = struct
         G.i (Convert (Wasm_exts.Values.I32 I32Op.WrapI64)) ^^
         G.i (Store {ty = I32Type; align = 0; offset = 0L; sz = None}) in
 
+      (* See comment on 64-bit destabilization on Note [mutable stable values] *)
+      let pointer_compression_shift = 3L in (* log2(word_size), 3 unused lower bits with 64-bit alignment *)
+      
+      let write_compressed_pointer env =
+        let (set_pointer, get_pointer) = new_local env "pointer" in
+        compile_add_const ptr_unskew ^^
+        compile_shrU_const pointer_compression_shift ^^
+        compile_add_const ptr_skew ^^
```

Comment:
Curious, why skew the compressed pointer?

Reply:
The reason why I skew the compressed pointer is only because the read operation would otherwise need to handle the (unskewed) zero case that denotes an empty memo (first occurrence of that shared object). Currently, due to the unskew/skew pair in the read operation, the zero sentinel remains unchanged.
Of course, we could also keep the unskewed compressed pointer and treat the empty case specifically. Alternatively, the write could even shift without skewing (chopping off  `0b111` bits) and the read operation could shift back and paste in the the lower `0b111`.

Reply:
Whatever works (just keep as is)